src/Microsoft.VisualStudio.Services.Agent/Util/WhichUtil.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.VisualStudio.Services.Agent.Util
+{
+    [ServiceLocator(Default = typeof(WhichUtil))]
+    public interface IWhichUtil : IAgentService
+    {
+        string Which(string command);
+    }
+
+    public sealed class WhichUtil : AgentService, IWhichUtil
```

Comment:
since it is AgentService now, it will be great to add some Trace during which().